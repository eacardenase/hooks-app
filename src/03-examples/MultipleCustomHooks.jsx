import { useFetch, useCounter } from '../hooks';

import { Quote, LoadingQuote } from './';

export const MultipleCustomHooks = () => {
    const { counter, incrementValue } = useCounter(1);

    const { data, isLoading, hasError } = useFetch(
        `https://www.breakingbadapi.com/api/quotes/${counter}`
    );

    const { author, quote } = !!data && data[0];

    // if (isLoading) {
    //     return <div className="alert alert-info text-center">Loading...</div>;
    // }

    return (
        <>
            <h1>Braking Bad Quotes</h1>
            <hr />

            {isLoading ? (
                <LoadingQuote />
            ) : (
                <Quote author={author} quote={quote} />
            )}

            <button
                className="btn btn-primary"
                onClick={() => incrementValue()}
                disabled={isLoading}
            >
                Next quote
            </button>
        </>
    );
};
