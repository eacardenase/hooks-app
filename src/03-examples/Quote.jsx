import PropTypes from 'prop-types';

export const Quote = ({ author, quote }) => {
    return (
        <>
            <blockquote className="blockquote text-end">
                <p className="mb-1">{quote}</p>
                <footer className="blockquote-footer mt-1">{author}</footer>
            </blockquote>
        </>
    );
};

Quote.propTypes = {
    quote: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
};
