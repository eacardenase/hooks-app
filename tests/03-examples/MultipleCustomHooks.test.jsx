import { cleanup, render, screen } from '@testing-library/react';
import { describe, test, expect, beforeEach } from 'vitest';
import { MultipleCustomHooks } from '../../src/03-examples/MultipleCustomHooks';

describe('<MultipleCustomHooks />', () => {
    beforeEach(cleanup);

    test('should match snapshot', () => {
        const { container } = render(<MultipleCustomHooks />);

        expect(container).toMatchSnapshot();
    });

    test('should show default component', () => {
        render(<MultipleCustomHooks />);

        const nextButton = screen.getByRole('button', { name: 'Next quote' });

        expect(screen.getByText('Loading...'));
        expect(screen.getByText('Breaking Bad Quotes'));
        expect(nextButton).toBeDefined();
        expect(nextButton.disabled).toBeTruthy();
    });
});
