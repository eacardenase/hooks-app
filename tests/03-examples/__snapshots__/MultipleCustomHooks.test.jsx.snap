// Vitest Snapshot v1

exports[`<MultipleCustomHooks /> > should match snapshot 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    class="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    class="btn btn-primary"
    disabled=""
  >
    Next quote
  </button>
</div>
`;
